pubsub.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"google.golang.org/cloud/pubsub"
+)
+
+func createPubSubClient(projectID string) *pubsub.Client {
+	client, err := pubsub.NewClient(context.Background(), projectID)
+	if err != nil {
+		log.Fatal("Failed to create Pub/Sub client", err)
+	}
+
+	return client
+}
+
+func parseProjectAndSubscriptionIDs(source string) (string, string, error) {
+	items := strings.Split(source, "/")
+	if len(items) != 2 {
+		return "", "", fmt.Errorf("Can not parse pubsub source: %s", source)
+	}
+
+	return items[0], items[1], nil
+}
+
+func runPubSubPoller(source string, minfoChan chan *metricInfo) error {
+	projectID, subID, err := parseProjectAndSubscriptionIDs(source)
+	if err != nil {
+		return err
+	}
+
+	sub := createPubSubClient(projectID).Subscription(subID)
+	it, err := sub.Pull(context.Background())
```

Comment:
pass in context from function. That context can be cancellable from main and let you do a clean shutdown.